src/Microsoft.ML.FastTree/Utils/LinqExtensions.cs

```diff
@@ -36,6 +36,19 @@ public static class LinqExtensions
             return argMax;
         }
 
+        public static int ArgMax<T>(this ReadOnlySpan<T> span) where T : IComparable<T>
```

Comment:
You should be able to remove the method above this, since arrays are implicitly convertible to ReadOnlySpan.